src/components/ItemProvider/ItemProvider.tsx

```diff
@@ -3,15 +3,16 @@ import { Props } from './ItemProvider.types'
 
 export default class ItemProvider extends React.PureComponent<Props> {
   componentDidMount() {
-    const { id, item, onFetchItem } = this.props
-    if (id && !item) {
+    const { id, item, onFetchItem, isConnected } = this.props
+    if (isConnected && id && !item) {
       onFetchItem(id)
     }
   }
 
   componentDidUpdate(prevProps: Props) {
-    const { id, item, onFetchItem } = this.props
-    if (id && id !== prevProps.id && !item) {
+    const { id, item, onFetchItem, isConnected } = this.props
+    const idChanged = id && id !== prevProps.id
+    if (isConnected && id && (idChanged || !item)) {
```

Comment:
@cazala I'm not sure the `idChanged` in here is needed. The `&& !item` might be enough. Is there away to navigate between items from the detail page? I want to simulate the `id` prop changing without unmounting the component

Reply:
mm i don't think you can navigate between two item detail pages directly, but this `ItemProvider` might be used somewhere else too. The purpose of this component is to recibe an id and provide an item to its children. I think the reason it still fetches an item even if it is already in memory is to update it (in case there were changes since the last time it was fetched).